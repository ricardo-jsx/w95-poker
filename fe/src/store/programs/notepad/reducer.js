const INITIAL_STATE = {
  content: '',
  isOpen: false,
};

export default function reducer(state = INITIAL_STATE, action) {
  console.log(action);
  switch (action.type) {
    case 'OPEN_NOTEPAD': {
      return { ...state, isOpen: true, content: action.payload.content };
    }

    default:
      return state;
  }
}

export const isNotepadOpen = (state) => state.programs.notepad.isOpen;
