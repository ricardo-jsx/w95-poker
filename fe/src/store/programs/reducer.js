import { combineReducers } from 'redux';

import notepad from './notepad/reducer';

export default combineReducers({ notepad });
