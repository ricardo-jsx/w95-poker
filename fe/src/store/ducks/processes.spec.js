import React from 'react';

import reducer, * as actions from './processes';

const INITIAL_STATE = {
  running: [
    { id: '123', name: 'Notepad', minimized: true },
    { id: '456', name: 'Poker' },
  ],
  executionOrder: [
    { id: '456', name: 'Poker' },
    { id: '123', name: 'Notepad' },
  ],
};

describe('store/ducks/processes', () => {
  it('should start a new process', () => {
    const state = reducer(INITIAL_STATE, actions.startProcess('789', 'Solietaire', 'base64'));
    expect(state).toMatchSnapshot();
  });

  it('should display a minimized process', () => {
    const state = reducer(INITIAL_STATE, actions.displayProcess('123'));
    expect(state).toMatchSnapshot();
  });

  it('should minimize a visible process', () => {
    const state = reducer(INITIAL_STATE, actions.minimizeProcess('456'));
    expect(state).toMatchSnapshot();
  });

  it('should maximize a visible process', () => {
    const state = reducer(INITIAL_STATE, actions.maximizeProcess('456'));
    expect(state).toMatchSnapshot();
  });

  it('should close a process', () => {
    const state = reducer(INITIAL_STATE, actions.closeProcess('123'));
    expect(state).toMatchSnapshot();
  });
});
