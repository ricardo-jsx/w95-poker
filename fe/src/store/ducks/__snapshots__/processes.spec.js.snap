// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/ducks/processes should close a process 1`] = `
Object {
  "executionOrder": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
  ],
  "running": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
  ],
}
`;

exports[`store/ducks/processes should display a minimized process 1`] = `
Object {
  "executionOrder": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
    Object {
      "id": "123",
      "name": "Notepad",
    },
  ],
  "running": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
    Object {
      "id": "123",
      "minimized": false,
      "name": "Notepad",
    },
  ],
}
`;

exports[`store/ducks/processes should maximize a visible process 1`] = `
Object {
  "executionOrder": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
    Object {
      "id": "123",
      "name": "Notepad",
    },
  ],
  "running": Array [
    Object {
      "id": "123",
      "minimized": true,
      "name": "Notepad",
    },
    Object {
      "id": "456",
      "maximized": true,
      "minimized": false,
      "name": "Poker",
    },
  ],
}
`;

exports[`store/ducks/processes should minimize a visible process 1`] = `
Object {
  "executionOrder": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
    Object {
      "id": "123",
      "name": "Notepad",
    },
  ],
  "running": Array [
    Object {
      "id": "456",
      "minimized": true,
      "name": "Poker",
    },
    Object {
      "id": "123",
      "minimized": true,
      "name": "Notepad",
    },
  ],
}
`;

exports[`store/ducks/processes should start a new process 1`] = `
Object {
  "executionOrder": Array [
    Object {
      "id": "456",
      "name": "Poker",
    },
    Object {
      "id": "123",
      "name": "Notepad",
    },
    Object {
      "icon": "base64",
      "id": "789",
      "name": "Solietaire",
    },
  ],
  "running": Array [
    Object {
      "id": "123",
      "minimized": true,
      "name": "Notepad",
    },
    Object {
      "id": "456",
      "name": "Poker",
    },
    Object {
      "icon": "base64",
      "id": "789",
      "name": "Solietaire",
    },
  ],
}
`;
